src/Symfony/Component/Notifier/Bridge/Octopush/OctopushTransport.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\Octopush;
+
+use Symfony\Component\Notifier\Exception\TransportException;
+use Symfony\Component\Notifier\Exception\UnsupportedMessageTypeException;
+use Symfony\Component\Notifier\Message\MessageInterface;
+use Symfony\Component\Notifier\Message\SentMessage;
+use Symfony\Component\Notifier\Message\SmsMessage;
+use Symfony\Component\Notifier\Transport\AbstractTransport;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+final class OctopushTransport extends AbstractTransport
+{
+    protected const HOST = 'www.octopush-dm.com';
+
+    private $userLogin;
+    private $apiKey;
+    private $smsSender;
+    private $smsType;
+
+    public function __construct(string $userLogin, string $apiKey, string $from, string $smsType, HttpClientInterface $client = null, EventDispatcherInterface $dispatcher = null)
+    {
+        $this->userLogin = $userLogin;
+        $this->apiKey = $apiKey;
+        $this->smsSender = $from;
+        $this->smsType = $smsType;
+
+        parent::__construct($client, $dispatcher);
+    }
+
+    public function __toString(): string
+    {
+        return sprintf('octopush://%s?from=%s&sms_type=%s', $this->getEndpoint(), $this->smsSender, $this->smsType);
+    }
+
+    public function supports(MessageInterface $message): bool
+    {
+        return $message instanceof SmsMessage;
+    }
+
+    protected function doSend(MessageInterface $message): SentMessage
+    {
+        if (!$message instanceof SmsMessage) {
+            throw new UnsupportedMessageTypeException(__CLASS__, SmsMessage::class, $message);
+        }
+
+        $requestParameters = [
+            'user_login' => $this->userLogin,
+            'api_key' => $this->apiKey,
+            'sms_text' => $message->getSubject(),
+            'sms_recipients' => $message->getPhone(),
+            'sms_sender' => $this->smsSender,
+            'sms_type' => $this->smsType,
+        ];
+
+        $endpoint = sprintf('https://%s/api/sms/json?%s', $this->getEndpoint(), http_build_query($requestParameters));
```

Comment:
Given we send a `POST` request, I think it will be better to send the data in the BODY instead of the queryString